Adds the Mothroach

Justice12354 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Yup. That's pretty much it. This PR adds the Mothroach to the game, described as "An ancient ancestor of the moth that surprisingly looks like the crossbreed of a moth and a cockroach."

**Do you love the Mothroach?** Then you can cuddle with it and pat it, as well as place it on your head for extra cuteness.
**What if you hate it, though?** You can always kill and butcher Mothroaches in order to mass produce moth plushes for your own profit... How fun!

Either way, you win!

The Mothroach can be picked up and has a special on-head sprite (which looks really cute). It is able to vent-crawl and you may get one by randomly summoning a friendly mob through the gold slime extracts, or by ordering one through the Cargo Requests. After butchered, you may use its hide, a heart, and some cloth to craft a moth plushie, the most devilish of Devil's designs.

Full Preview of all the Sprites (NEW): https://www.youtube.com/watch?v=pdg8FTNEYjI
Preview of some of the Sprites (OLD): https://www.youtube.com/watch?v=9A-8hGCiW0s
In-hand, on-head, and grounded  Mothroach sprite credits go to **ValuedEmployee**.
I did the Mothroach hide sprite though!
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
The Mothroach is incredibly cute and a neat, fresh, new piece of content. Although it could use some future repurposing, right now it's simply a cute exotic pet with a few interactions.

These cute sprites are just too good to go to waste...

I keep seeing people complain about the lack of new content. Well, here's something niche that won't break the whole balance of the game and that will be cute. I seriously cannot see a motive not to add this to the game. Just because it isn't a powergaming tool or something that is seen every shift, that doesn't mean that it won't have a positive influence on the game. As I have stated, right now the Mothroaches are underperforming in terms of interactions and ways of getting them, but adding them is the first step to later improve them.
<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: The Mothroach, your new local exotic pet
add: Mothroach Hide and Mothroach Meat
add: New crafting recipe for the Moth Plush: 1 Mothroach Hide; 1 heart; 3 cloth
fix: Fixes dead mobs on-head not having sprites
/:cl:
<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Justice12354 committed:
mothroach

Justice12354 committed:
Adding the on-head sprite

Justice12354 committed:
A few tweaks

Justice12354 committed:
Mothroaches were dying to room temperature lol

Changes cold temp sensitivity from 300K to 250K

Justice12354 committed:
Can now be spawned through gold slime

OneAsianTortoise commented:
Sometimes I just want to stab my own eyes

Jacquerel commented:
It’s very cute but I do wish it did something other than wander randomly and be cute, even if it were just “tries to eat loose clothing”.

Justice12354 commented:
> It’s very cute but I do wish it did something other than wander randomly and be cute, even if it were just “tries to eat loose clothing”.

"occasionally tides into the AutoDrobe and eats everything" haha
I'm afraid I don't know how to do that, though I'd love to, if I could

Justice12354 committed:
No more nuke and bomb immunity

dragomagol commented:
Love the sprites for it riding on your head.

NamelessFairy commented:
needs to make the funny moth sound

Justice12354 commented:
> Love the sprites for it riding on your head.

Absolutely

> needs to make the funny moth sound

It's coming!

Justice12354 committed:
Fixes dead mobs on-head not having sprites

Justice12354 committed:
Does a bunch of stuff

- Makes Mothroach a simple animal;
- Gives it funny moth sounds when punched;
- Allows it to be butchered;
- Adds Mothroach Hide, which can be used with 1 heart and 3 cloth to make Moth Plushie;
- Adds Mothroach meat;
- Polishes some corners around the Mothroach.

itsmeow commented:
> > It’s very cute but I do wish it did something other than wander randomly and be cute, even if it were just “tries to eat loose clothing”.
> 
> "occasionally tides into the AutoDrobe and eats everything" haha
> I'm afraid I don't know how to do that, though I'd love to, if I could

better idea, add a station event where loads of clothing items around the map disappear (dorms lockers, vending machines, etc) and then just spawn mothroaches all over after that and have it say "an infestation of cloth-munching creatures has migrated" or something

Justice12354 committed:
Fixes some runtimes and adds fluttering

Justice12354 commented:
> better idea, add a station event where loads of clothing items around the map disappear (dorms lockers, vending machines, etc) and then just spawn mothroaches all over after that and have it say "an infestation of cloth-munching creatures has migrated" or something

This sounds far more realistic, but I believe that this PR already has enough content and I honestly can't put the effort, right now, to figure out how events work and actually create a new one. As I said in the PR itself, this new animal may have some future repurposing in order to give it more interesting interactions, such as, for instance, the event you described.

itsmeow commented:
> > better idea, add a station event where loads of clothing items around the map disappear (dorms lockers, vending machines, etc) and then just spawn mothroaches all over after that and have it say "an infestation of cloth-munching creatures has migrated" or something
> 
> This sounds far more realistic, but I believe that this PR already has enough content and I honestly can't put the effort, right now, to figure out how events work and actually create a new one. As I said in the PR itself, this new animal may have some future repurposing in order to give it more interesting interactions, such as, for instance, the event you described.

I'd be willing to add something like this!

ArcaneMusic commented:
As it stands, these don't have the ability to spawn. correct? Through Gold Slimes, Cytology, map events?

Justice12354 commented:
> As it stands, these don't have the ability to spawn, correct? Through gold slimes, cytology, map events?

Only through gold slime extracts.

Justice12354 committed:
Adds petting bonus to mothroach

Don't you just feel happier when you pet such a cute little creature?

Werenimal commented:
I love mothroach. 
Would be better if it ate lose clothes. Like if someone drops a clothing item they all rush over to it to eat it. That would be great.

Justice12354 committed:
Back to basic mob and fix sounds

Justice12354 committed:
When the indentation gives you a brain dent

Addust commented:
merge this

JohnFulpWillard commented on code/datums/components/crafting/recipes.dm:
```diff
@@ -669,6 +669,14 @@
+	result = /obj/item/toy/plush/moth
+	reqs = list(/obj/item/stack/sheet/animalhide/mothroach = 1,
+				/obj/item/organ/internal/heart = 1,
+				/obj/item/stack/sheet/cloth/ = 3)
```
```suggestion
				/obj/item/stack/sheet/cloth = 3)
```

JohnFulpWillard commented on code/datums/components/crafting/recipes.dm:
```diff
@@ -669,6 +669,14 @@
 	blacklist |= typesof(/obj/item/radio/headset)
 	blacklist |= typesof(/obj/item/radio/intercom)
 
+/datum/crafting_recipe/mothplush
```
this crafting recipe is entirely unmentioned in the changelog, also.

Justice12354 replied:
I would disagree
"Mothroach hide, which can be used to craft a moth plush, and Mothroach meat"
I can be more specific about it, but it's not "entirely unmentioned"

Justice12354 replied:
Done

Justice12354 replied:
Done

DaydreamIQ commented:
I legitimately wanna see this as a pAI skin now

Justice12354 committed:
Update code/datums/components/crafting/recipes.dm

Co-authored-by: John Willard <53777086+JohnFulpWillard@users.noreply.github.com>

Justice12354 commented:
> I legitimately wanna see this as a pAI skin now

I am not gonna recolor it into a pAI skin. You can ask the creator of the mothroach, though.

Jacquerel commented:
Rather than being obtainable primarily through gold slime extracts, if these are intended as cute player pets would it make sense to be able to buy a crate containing a single mothroach through cargo personal orders?

Justice12354 commented:
> Rather than being obtainable primarily through gold slime extracts, if these are intended as cute player pets would it make sense to be able to buy a crate containing a single mothroach through cargo personal orders?

I do feel like there is currently a lack of ways to obtain a mothroach (hence why I spawn them often). I do feel like creating a cargo crate for them would be a great idea. Tomorrow I will work on that. Thanks for the feedback.

Justice12354 committed:
Typo fixed. Mothroach can be bought from Cargo

Fikou commented on code/modules/mob/living/basic/vermin/mothroach.dm:
```diff
@@ -0,0 +1,66 @@
+	add_verb(src, /mob/living/proc/toggle_resting)
+	ADD_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+
+/mob/living/basic/mothroach/death(gibbed)
+	..()
+	attacked_sound = SFX_PUNCH
```
what does this do here

Justice12354 replied:
Basically, if it's not there, the mothroach will do the moth scream sound when hit, even if dead. 
By adding that, when the mothroach dies, it will swap the screaming for normal punching.

Fikou replied:
they should react to being hit with this sound, they shouldnt replace the hitting sound with it

Justice12354 replied:
I feel like having the moth scream + punching would be a bit loud or messy. I haven't tested it, but I can give it a try today or tomorrow.

Justice12354 replied:
I believe the new commit should cover any concerns you've got

Fikou commented on code/modules/mob/living/basic/vermin/mothroach.dm:
```diff
@@ -0,0 +1,66 @@
+	ai_movement = /datum/ai_movement/basic_avoidance
+	idle_behavior = /datum/idle_behavior/idle_random_walk
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/random_speech/mothroach
+)
```
indent this

Fikou replied:
```suggestion
		/datum/ai_planning_subtree/random_speech/mothroach,
	)
```

Mickyan commented on code/modules/mob/living/basic/vermin/mothroach.dm:
```diff
@@ -0,0 +1,66 @@
+/mob/living/basic/mothroach
+	name = "mothroach"
+	desc = "An ancient ancestor of the moth that surprisingly looks like the crossbreed of a moth and a cockroach."
```
Current moth lore does not permit precursor species to be present in the known world as the moth planet has been lost a long time ago, you could handwave this as being a product of genetic manipulation and it would be fine

Justice12354 replied:
What do you think of "This is the adorable by-product of multiple attempts at genetically mixing mothpeople with cockroaches."?

Mickyan replied:
Sounds horrifying and probably unethical, but no lore objections!

Justice12354 replied:
If you got any suggestions to make it sound more horrifying and unethical, I'm all for it!

Justice12354 committed:
Update code/modules/mob/living/basic/vermin/mothroach.dm

Co-authored-by: Fikou <23585223+Fikou@users.noreply.github.com>

Justice12354 committed:
Update desc

Justice12354 committed:
Merge branch 'mothroach' of https://github.com/Justice12354/tgstation into mothroach

scriptis commented:
If you aheal a dead mothroach, it keeps its dead sprite, despite reviving. Also, the lazarus injector doesn't work on them; not sure if that's intentional.

Justice12354 commented:
> If you aheal a dead mothroach, it keeps its dead sprite, despite reviving. Also, the lazarus injector doesn't work on them; not sure if that's intentional.

That's a basic mob bug. Try it on the cow or stickmen, and the same will happen. There is nothing I can do about that

Justice12354 committed:
mothroaches now scream when hit with batons

TerraGS commented:
I think mothroach is lacking a burning overlay, as one just crawled over me and lit me on fire.

Justice12354 commented:
> I think mothroach is lacking a burning overlay, as one just crawled over me and lit me on fire without appearing lit.

That's a known basic mob issue that's getting fixed in #68971
